<!--REQUEST_ID=fe3a5228-0435-4081-bb39-d7f23223a6ab-->
Timed out, unknown build status on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.50471&attempt_id=4ffa87de-5664-4a50-bfa6-5000299ccc2f)

Attempted: pythonPackages.fs-s3fs, python3Packages.fs-s3fs

<details><summary>Partial log (click to expand)</summary><p>

```
AssertionError: [psutil.Process(pid=2272, name='python2.7', started='03:04:55'), psutil.Process(pid=2274, name='python2.7', started='03:04:56'), psutil.Process(pid=2211, name='python2.7', started='03:04:53'), psutil.Process(pid=2276, name='python2.7', started='03:04:56'), psutil.Process(pid=2277, name='python2.7', started='03:04:56'), psutil.Process(pid=2278, name='python2.7', started='03:04:56'), psutil.Process(pid=2273, name='python2.7', started='03:04:56'), psutil.Process(pid=2280, name='python2.7', started='03:04:56'), psutil.Process(pid=2281, name='python2.7', started='03:04:56'), psutil.Process(pid=2279, name='python2.7', started='03:04:56'), psutil.Process(pid=2275, name='python2.7', started='03:04:56'), psutil.Process(pid=2266, name='python2.7', started='03:04:55'), psutil.Process(pid=2267, name='python2.7', started='03:04:55'), psutil.Process(pid=2268, name='python2.7', started='03:04:55'), psutil.Process(pid=2269, name='python2.7', started='03:04:55'), psutil.Process(pid=2270, name='python2.7', started='03:04:55'), psutil.Process(pid=2271, name='python2.7', started='03:04:55')]

----------------------------------------------------------------------
Ran 764 tests in 49.522s

FAILED (failures=6, errors=345, skipped=55)
building of '/nix/store/mvgdnsikmk6iqpwkgfc9mqj8vvfs8kj4-python2.7-pyftpdlib-1.5.4.drv' timed out after 3600 seconds
cannot build derivation '/nix/store/fa0s38038gjfc6cp6kng75cxslvgkm17-python2.7-fs-2.1.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/ni2hfcwq1hxgnry1dfz2wdb1m84r2ml0-python2.7-fs-s3fs-1.0.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/ni2hfcwq1hxgnry1dfz2wdb1m84r2ml0-python2.7-fs-s3fs-1.0.0.drv' failed
```
</p></details>

--- a/pkgs/development/python-modules/fs-s3fs/default.nix
+++ b/pkgs/development/python-modules/fs-s3fs/default.nix
@@ -1,0 +1,23 @@
+{ buildPythonPackage, fetchPypi, lib, fs, six, boto3 }:
+
+buildPythonPackage rec {
+  pname = "fs-s3fs";
+  version = "1.0.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1czv67zs4sl5l3rv9l3hzn22zzzqm372lq1wlhihigir4cfyslak";
+  };
+
+  propagatedBuildInputs = [ fs six boto3 ];
+
+  # tests try to integrate an s3 bucket which can't be tested properly in an isolated environment.
+  doCheck = false;
+
+  meta = with lib; {
+    homepage = https://pypi.org/project/fs-s3fs/;
+    license = licenses.mit;
+    description = "Amazon S3 filesystem for PyFilesystem2";
+    maintainers = with maintainers; [ ma27 ];
+  };
+}

--- a/pkgs/development/python-modules/pyftpdlib/default.nix
+++ b/pkgs/development/python-modules/pyftpdlib/default.nix
@@ -20,9 +20,9 @@ buildPythonPackage rec {
   checkInputs = [ mock psutil ];
   propagatedBuildInputs = [ pyopenssl pysendfile ];
 
-  checkPhase = ''
-    ${python.interpreter} pyftpdlib/test/runner.py
-  '';
+  # impure filesystem-related tests cause timeouts
+  # on Hydra: https://hydra.nixos.org/build/84374861
+  doCheck = false;
 
   meta = with stdenv.lib; {
     homepage = https://github.com/giampaolo/pyftpdlib/;

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -2658,6 +2658,8 @@ in {
 
   fs = callPackage ../development/python-modules/fs { };
 
+  fs-s3fs = callPackage ../development/python-modules/fs-s3fs { };
+
   libcloud = callPackage ../development/python-modules/libcloud { };
 
   libgpuarray = callPackage ../development/python-modules/libgpuarray {

